app/core/OAuthService/OAuthInterface.ts

```diff
@@ -0,0 +1,59 @@
+import { AuthSessionResult } from 'expo-auth-session';
+import { Web3AuthNetwork } from '@metamask/seedless-onboarding-controller';
+
+export type HandleOAuthLoginResult =
+  | { type: 'pending' }
+  | { type: AuthSessionResult['type']; existingUser: boolean }
+  | { type: 'error'; error: string };
+
+export enum AuthConnection {
+  Google = 'google',
+  Apple = 'apple',
+}
+
+export interface LoginHandlerCodeResult {
+  authConnection: AuthConnection;
+  code: string;
+  clientId: string;
+  redirectUri?: string;
+  codeVerifier?: string;
+}
+
+export interface LoginHandlerIdTokenResult {
+  authConnection: AuthConnection;
+  idToken: string;
+  clientId: string;
+  redirectUri?: string;
+  codeVerifier?: string;
+}
+
+export type LoginHandlerResult =
+  | LoginHandlerCodeResult
+  | LoginHandlerIdTokenResult;
+
+export type HandleFlowParams = LoginHandlerResult & {
+  web3AuthNetwork: Web3AuthNetwork;
+};
+
+export interface OAuthUserInfo {
+  email: string;
+  sub: string;
+}
+
+export interface AuthResponse {
+  id_token: string;
+  verifier: string;
+  verifier_id: string;
+  indexes: Record<string, number>;
+  endpoints: Record<string, string>;
```

Comment:
i believe this is reponse type of byoa request, don't think it returns all this endpoints and indexes etc.

Reply:
updated with latest interface